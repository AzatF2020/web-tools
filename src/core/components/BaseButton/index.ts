export { default as BaseButton } from './ui/BaseButton.tsx';
