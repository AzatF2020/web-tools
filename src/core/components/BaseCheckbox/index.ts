export type { IBaseCheckboxProps } from './types/types';

export { default as BaseCheckbox } from './ui/BaseCheckbox';
