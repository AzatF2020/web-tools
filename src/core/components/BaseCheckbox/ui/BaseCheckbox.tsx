import { forwardRef } from 'react';
import type { IBaseCheckboxProps } from '@/core/components/BaseCheckbox/types/types';
import styles from '@/core/components/BaseCheckbox/styles/BaseCheckbox.module.scss';
import clsx from 'clsx';
import '../styles/_checkbox-global.scss';

const BaseCheckbox = forwardRef<HTMLInputElement, IBaseCheckboxProps>(
  (
    {
      className = '',
      value = '',
      disabled = false,
      invalid = false,
      checked = false,
      onChange,
      ...props
    },
    ref,
  ) => {
    return (
      <div className={styles.checkbox}>
        <input
          {...props}
          value={value}
          checked={checked}
          disabled={disabled}
          onChange={onChange}
          type="checkbox"
          role="checkbox"
          ref={ref}
          className={clsx(styles.checkbox__component)}
          aria-invalid={invalid}
          aria-checked={checked}
        />
        <span
          className={clsx(
            styles.checkbox__checkmark,
            disabled && styles['checkbox__checkmark--disabled'],
            invalid && styles[`checkbox__checkmark--invalid`],
            className,
          )}></span>
      </div>
    );
  },
);

export default BaseCheckbox;
