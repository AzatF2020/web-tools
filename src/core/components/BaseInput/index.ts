export type { IBaseInputProps } from './types/types';

export { default as BaseInput } from './ui/BaseInput.tsx';
