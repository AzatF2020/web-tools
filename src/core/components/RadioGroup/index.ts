export type { IRadioItemProps } from './types/types';

export { default as RadioItem } from './ui/RadioItem';
