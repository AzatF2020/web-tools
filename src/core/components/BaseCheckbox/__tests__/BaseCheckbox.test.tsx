import { describe, it, expect, vi } from 'vitest';
import { screen, render, fireEvent } from '@testing-library/react';
import { BaseCheckbox, type IBaseCheckboxProps } from '@/core/components/BaseCheckbox';

const renderCheckbox = (props: IBaseCheckboxProps = {}) => {
  return render(<BaseCheckbox {...props} />);
};

const getCheckmark = (container: HTMLElement): HTMLSpanElement | null => {
  return container.querySelector('.checkbox__checkmark');
};

describe('BaseCheckbox test component', () => {
  it('should render with props', async () => {
    const onChange = vi.fn();
    const { container } = renderCheckbox({
      checked: true,
      invalid: true,
      disabled: false,
      value: 'checkbox testing',
      'aria-checked': false,
      'aria-invalid': false,
      onChange,
    });
    expect(container).toMatchSnapshot();
  });

  it('should render with disabled state and className', () => {
    const { container } = renderCheckbox({ disabled: true });

    const input = screen.getByRole('checkbox');
    const checkmark = getCheckmark(container);

    expect(input).toHaveAttribute('disabled');
    expect(checkmark).toHaveClass('checkbox__checkmark--disabled');
  });

  it('should test onChange function with correct value', async () => {
    const onChange = vi.fn();

    const { getByRole } = renderCheckbox({ onChange });

    const input = getByRole('checkbox');

    expect(input).not.toBeChecked();

    await fireEvent.change(input, { target: { checked: true } });

    expect(input).toBeChecked();
  });

  it('should test with incorrect attribute value', () => {
    const { getByRole } = renderCheckbox({ invalid: true, disabled: true });

    expect(getByRole('checkbox')).not.toHaveAttribute('aria-invalid', false);
    expect(getByRole('checkbox')).not.toHaveAttribute('disabled', false);
  });

  it('should test with new className', () => {
    const { container } = renderCheckbox({ className: 'checkbox-custom-class' });

    expect(getCheckmark(container)).toHaveClass('checkbox-custom-class');
  });
});
