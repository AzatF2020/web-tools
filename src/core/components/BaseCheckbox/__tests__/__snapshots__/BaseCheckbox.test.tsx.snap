// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseCheckbox test component > should render with props 1`] = `
<div>
  <div
    class="checkbox"
  >
    <input
      aria-checked="true"
      aria-invalid="true"
      checked=""
      class="checkbox__component"
      role="checkbox"
      type="checkbox"
      value="checkbox testing"
    />
    <span
      class="checkbox__checkmark checkbox__checkmark--invalid"
    />
  </div>
</div>
`;
